src/opera/commands/notify.py

```diff
@@ -0,0 +1,141 @@
+    )
+    parser.add_argument(
+        "--notification", "-n", type=argparse.FileType("r"),
+        help="Notification file (usually JSON) with changes that will be exposed to TOSCA interfaces",
+    ).complete = shtab.FILE
+    parser.add_argument(
+        "--force", "-f", action="store_true",
+        help="Force the action and skip any possible prompts",
+    )
+    parser.add_argument(
+        "--verbose", "-v", action="store_true",
+        help="Turns on verbose mode",
+    )
+    parser.set_defaults(func=_parser_callback)
+
+
+def _parser_callback(args):
+    if args.instance_path and not path.isdir(args.instance_path):
+        raise argparse.ArgumentTypeError("Directory {} is not a valid path!".format(args.instance_path))
+
+    storage = Storage.create(args.instance_path)
+    status = info(None, storage)["status"]
+
+    if not args.force and storage.exists("instances"):
+        if status == "initialized":
+            print("The nodes have not been deployed yet and running notify might have unexpected consequences.")
+            question = prompt_yes_no_question()
+            if not question:
+                return 0
+        if status == "interrupted":
+            print("The deployment has been interrupted and running notify might have "
+                  "unexpected consequences on the nodes that have not yet been deployed.")
+            question = prompt_yes_no_question()
+            if not question:
+                return 0
+        if status == "undeployed":
+            print("All nodes have been undeployed and running notify might have unexpected consequences.")
+            question = prompt_yes_no_question()
+            if not question:
+                return 0
+
+    if not args.force and not args.trigger:
+        print("You have not specified which policy trigger to use (with --trigger/-t or --event/-e) "
+              "and in this case all the triggers will be invoked which might not be what you want.")
+        question = prompt_yes_no_question()
+        if not question:
+            return 0
+
+    # read the notification file and the pass it's contents to the library function
+    notification_file_contents = Path(args.notification.name).read_text() if args.notification else None
+
+    try:
+        if args.inputs:
+            inputs = yaml.safe_load(args.inputs)
+        else:
+            inputs = None
+    except YAMLError as e:
+        print("Invalid inputs: {}".format(e))
+        return 1
+
+    try:
+        notify(inputs, storage, args.verbose, args.trigger, notification_file_contents)
+    except ParseError as e:
+        print("{}: {}".format(e.loc, e))
+        return 1
+    except DataError as e:
+        print(str(e))
+        return 1
+
+    return 0
+
+
+def notify(inputs: typing.Optional[dict], storage: Storage, verbose_mode: bool, trigger_name_or_event: str,
+           notification_file_contents: typing.Optional[str]):
```

Comment:
The signature here has `trigger_name_or_event`, but you don't have a `--event` argument. Intentional?

Reply:
Yes it's intentional to indicate that trigger OR event name should go here. At first I had a separated options for ``--trigger/-t`` and ``--event`` but then I changed my mind because every trigger always has its event in TOSCA. But since event is defined as `The required name of the event that activates the trigger’s action. A deprecated form of this keyname is “event_type”. `, I was not completely sure that only events should be used to filter out triggers, so I added the option to use also the name of the triggers which is probably even better (because YAML will find duplicate trigger names and will prevent that).

Reply:
If an event and a trigger have the same name, what happens? My concern here is that we're obscuring this already confusing notification/event/trigger terminology even further. If you want to keep this, I'd suggest adding an `--event` alias to the command line argument.

But I'd much rather only have one way of specifying what gets called, because it's simpler to explain. If we keep both, we blur the disinction between event names and trigger names, which makes understanding things harder. That's just my gut feeling though, I don't write TOSCA :)

Reply:
A trigger can have the same name as its event and nothing will happen. Each trigger always has an event that activates it. As TOSCA states event represents the required name associated with an interface notification on the identified resource (node).

![image](https://user-images.githubusercontent.com/41727816/112836506-fb468c00-909a-11eb-9a58-346a9765717a.png)

So there's much more meaning in those events and surely we cannot say that trigger is the same as event. But when invoking triggers I don't see the actual reason to distinct these two parameters. So, naturally events invoke triggers and the same would be in SaaS, but as a user I would also want to invoke triggers by their name maybe because some event would activate multiple triggers and I would want just want so I would specify it exactly by its name. Or there could be cases when the user would want to invoke multiple triggers by their name prefix (like all RADON triggers that start with radon.triggers). So I suggest that we keep both options but we focus on the actual trigger events more. If we need to remove trigger name filtering we can do it anytime. I have renamed `--trigger/-t` to `--event/-e`.